### DISCUSSION TITLE
Suggestion is only ever "// your code here return 0; }" ?

### DISCUSSION THREAD

**Discussion by drscotthawley** [community/community#45501] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

The suggested completions I see in VSCode do not match what I see from others.  No matter what I type, the only completions I see are the extremely boring 
```javascript
  // your code here
  return 0;
}
```

For example, in the [QuickStart](https://docs.github.com/en/copilot/quickstart), we are suggested to start with 
```javascript
function calculateDaysBetweenDates(begin, end) {
```

and thus I get a function that is 

```javascript
function calculateDaysBetweenDates(begin, end) {
  // your code here
  return 0;
}
```

If I write other functions -- with very descriptive names -- the suggested code is the same. Here are a few examples:
```javascript
function bubbleSort(array) {
  // your code here
  return array;
}
```

```javascript
function generateFibonacciSequence(n) {
  // your code here
  return [];
}
```
??

How can I help improve the quality of your code suggestions?  in other words, what is the point of an assistant that only ever tells me "your code here" and "return <thing>"?

**Comment by drscotthawley** [comment]
Also, I note that Option-]  (I'm on Mac) does not cause any new suggestions to be given.

**Reply by jge162** [reply]
option + ] shortcut is used to navigate between matching bracket pairs in Visual Studio Code. It will not provide new code suggestions.

what you can try to do @drscotthawley is install IntelliSense feature in Visual Studio Code. IntelliSense provides suggestions for code completion, parameter info, and quick info as you type. The suggestions will appear in a dropdown menu as you type. You can trigger IntelliSense suggestions by typing or by pressing "Cmd + Space" on mac only. 

furthermore,  regarding suggestions you are seeing are likely coming from a code snippet that is built into Visual Studio Code or an extension you have installed. You can customize or remove these snippets by going to the settings editor in VS Code (File > Preferences > Settings) and searching for "snippets". see screen shot...

<img width="862" alt="Screen Shot 2023-01-26 at 9 46 01 PM" src="https://user-images.githubusercontent.com/31228460/215018220-1232392b-a60c-493d-8e9e-936b038b6eac.png">

**Reply by drscotthawley** [reply]
@jge162 thank you for your answer.  As for your first statement, 
> option + ] shortcut is used to navigate between matching bracket pairs in Visual Studio Code. It will not provide new code suggestions.

...? But that is literally what it says to do in the documentation for [Seeing alternative suggestions](https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-visual-studio-code#seeing-alternative-suggestions):
<img width="782" alt="Screen Shot 2023-01-26 at 11 51 53 PM" src="https://user-images.githubusercontent.com/13925685/215018847-8f079c8b-5dea-4799-ad63-d8e788ebf079.png">
So your statement seems to contradict the documentation. 

Re. Intellisense: I can't see any mention of that in the [Getting started with GitHub Copilot in Visual Studio Code](https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-visual-studio-code#seeing-your-first-suggestion) page.  It shows, without any mention of Intellisense, a code suggestion that looks like the following: 

![getting started code suggestion](https://docs.github.com/assets/cb-44686/images/help/copilot/first-suggestion-visual-studio-code.png). 

>  IntelliSense provides suggestions for code completion, parameter info, and quick info as you type. The suggestions will appear in a dropdown menu as you type. You can trigger IntelliSense suggestions by typing or by pressing "Cmd + Space" on mac only.

It sounds like you are answering a different question than I'm asking.  I'm interested in trying to use Copilot.  Hence the "copilot" tag for this page. It sounds as if you are describing an alternative piece of software that performs the same functionality as Copilot. But my issue is specifically about using Copilot, not asking for alternatives. 


Re. Snippets.  Thank you. Again it is very strange that I see no mention of "snippets" in the "GitHub Copilot + VSCode" documentation.  But now that I have gone through and turned off all the snippet options, I see *no* code suggestion at all!   If I go back to Extensions I see that GitHub Copilot is still Enabled though.  If I look in the bottom right there is an icon for "Deactivate Copilot" which implies Copilot is enabled.  Yet, no code suggestions at all.... 

So, now I seem to be worse off than when I started.  🤔

**Reply by jge162** [reply]
I failed to see that the question was directed in copilot. (my mistake). You did not mention copilot in your (body) so I overlooked it.

**Reply by jge162** [reply]
If you're not seeing any code suggestions at all. you can also try disabling all other extensions and then re-enabling them one by one to see if any of them are causing conflicts with code suggestions.

**Reply by jge162** [reply]
re: snippets, It was only mentioned as an oversight. ( I missed the "copilot" tag)

**Reply by drscotthawley** [reply]
I have now disabled all extensions except for GitHub Copilot.  
No change.  Same boring suggestions. 
I exited and restarted VSCode.  No change. 

Suggestion look likes this, instead of above.  And all other suggestions look (basically) like this :

<img width="579" alt="Screen Shot 2023-01-27 at 12 24 43 AM" src="https://user-images.githubusercontent.com/13925685/215022825-bd224c56-f2e9-4f29-a39e-f7706f50c5fd.png">

**Reply by jge162** [reply]
<img width="489" alt="Screen Shot 2023-01-26 at 10 29 55 PM" src="https://user-images.githubusercontent.com/31228460/215023474-b4162acc-2039-4f1a-831d-552c3e850ddc.png">

my suggestions are like this, is this more what you are looking for?

**Reply by jge162** [reply]
<img width="463" alt="Screen Shot 2023-01-26 at 10 34 19 PM" src="https://user-images.githubusercontent.com/31228460/215024097-ba68404e-9a8c-4959-98d7-49b9aa3bb3c1.png">

**Comment by jge162** [comment]
have you tried doing this yet? @drscotthawley 
<img width="641" alt="Screen Shot 2023-01-26 at 10 40 42 PM" src="https://user-images.githubusercontent.com/31228460/215025017-f6789a8c-13ec-4413-bd57-57f70d87991f.png">
